### DISCUSSION TITLE
x-on shorthand does not work with custom prefix

### DISCUSSION THREAD

**Discussion by doompadee** [alpinejs/alpine#2186] [7. Bugs (archived)] [answered]
The shorthand notation for the `x-on` directive does not work when a custom attribute prefix is configured.

With

```javascript
Alpine.prefix("data-x-");
```
event handlers fail:

```html
<button
   data-x-cloak
   data-x-show="nextBtn"
   @click="revealNext()">Next</button>
```

This is a pity as the extended syntax is not HTML5 compliant (see https://html.spec.whatwg.org/#embedding-custom-non-visible-data-with-the-data-*-attributes).

**Comment by SimoTod** [comment, marked as answer]
Attributes starting with @ are not complaints anyway at the moment the only way is to use x-bind and define directives with colons inside the Alpine data

**Reply by doompadee** [reply]
Thanks. Silly me, the error messages for @click etc were filtered out during my testing...

Do you have an example for the suggested usage of x-bind? Sounds complicated. Can this style still be done within the HTML element only?

**Reply by SimoTod** [reply]
I don't have an example but you can find the docs here: https://alpinejs.dev/directives/bind#bind-directives

Another user interested in having the site w3c valid used it.

**Reply by doompadee** [reply]
Got it! Indeed, now there are no more errors regarding events. A bit of a burden to implement, but at least there is a way to stay compliant.

Thanks very much for your help. Keep up the good work!
